hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/OmUtils.java

```diff
@@ -812,4 +813,9 @@ public static String getOMAddressListPrintString(List<OMNodeDetails> omList) {
     printString.append("]");
     return printString.toString();
   }
+
+  // Key points to entire bucket's snapshot
+  public static boolean isBucketSnapshotIndicator(String key) {
+    return key.startsWith(OM_SNAPSHOT_INDICATOR) && key.split("/").length == 2;
```

Comment:
+1

Reply:
The OzoneBucket.listKeys() method has a keyPrefix parameter.  The parameter can be null/empty, and invoked like so:
```
   bucket.listKeys("");
```
If that paramter is null or empty, it is not supposed to be included in the listKeys result.  [This is how the code looked prior to me adding snapshot reads](https://github.com/apache/ozone/blob/master/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java#L1370-L1375)

The isBucketSnapshotIndicator() method tests to see if we have the snapshot equivalent of an empty keyPrefix. The indicator looks like so: ".snapshot/snapshot1", so the empty listKeys() invocation for snapshots would look like:
```
   bucket.listKeys(".snapshot/snapshot1");
```

The isBucketSnapshotIndicator() method is only used by ozoneBucket.listKeys() [here](https://github.com/apache/ozone/blob/acc7e5e0e870e2f6a6aa1b1570cb95c3a73ddf3d/hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java#L1376).

So maybe it would be clearer/better, if I just made it a private method in that class, and called it "isSnapshotEmptyPrefix()".  Would that be clearer?

WRT "sanitizing", I was not concerned about extra "/"'s at the end of the string, because the code doesn't sanitize the empty string "".  Is that a reasonable assumption?

Reply:
Resolving this comment here. If we find bugs around this, we can issue it in a separate PR.